dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -103,6 +107,27 @@ def check_unique_records(self, config: DimensionMappingsConfig, warn_only=False)
                     f"There are {len(duplicates)} dimension mappings with duplicate content (data files)."
                 )
 
+    def _check_records_against_dimension_records(self, config):
+        for mapping in config.model.mappings:
+            actual_from_records = {x.from_id for x in mapping.records}
+            actual_to_records = {x.to_id for x in mapping.records}
+            from_dimension = self._dimension_mgr.get_by_id(mapping.from_dimension.dimension_id)
+            allowed_from_records = from_dimension.get_unique_ids()
+            diff = actual_from_records.symmetric_difference(allowed_from_records)
+            if diff:
+                dim_id = from_dimension.model.dimension_id
+                raise DSGInvalidDimensionMapping(
+                    f"Dimension mapping has invalid 'from' records: dimension_id={dim_id} {diff}"
+                )
+            to_dimension = self._dimension_mgr.get_by_id(mapping.to_dimension.dimension_id)
+            allowed_to_records = to_dimension.get_unique_ids()
+            diff = actual_to_records.symmetric_difference(allowed_to_records)
```

Comment:
Sorry, is this only used in cases in which all 'to' dimension records are required? 

I think we landed on the notion that all mappings should list all from records, but may subset relative to the to-dimension. This looks like a pretty low-level function--should it have a kwarg to specify whether this check should be symmetric or should instead only look for invalid record ids?

Reply:
Presumably the association table aware-subsetting of the to-dimension will happen elsewhere (and in a different PR?)

Reply:
I fixed the error messages. The purpose of this PR is to fix a very specific issue. While it's possible that this function could be used for some other purpose, that is beyond the scope of what I want to do here.

Reply:
If `to_id` can be a subset of the registered dim it's assigned to, then it might lead to the following case:
Suppose we have this mapping:
```
from_id -> to_id
A->i
B->i
C->j
D->k
```

If Project 1 wants {I, j, k, l} for the to_dimension and Project 2 wants {i, j, k, l, m, n}. What should the `to_id` be referred to in the mapping? 
Should each project registers the same mapping but with different to_id? Or should the mapping have its own `to_id` dimension registered and have an association to project?

Reply:
Lixi - that case is actually something we will want. In that case with the two i's, we would need to aggregate them together.